### ISSUE TITLE
UPDATE statements with RETUR

### ISSUE THREAD

**Issue by ClayBenson94** [sqlfluff/sqlfluff#3195]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I was running `sqlfluff` on a .sql file that contained some Postgres-dialect SQL (`UPDATE ... RETURNING` statement) and got a parsing error. This only seems to happen on `UPDATE...RETURNING` and not on `INSERT...RETURNING`

### Expected Behaviour

SQLFluff parses the file as it does for a file with an `INSERT, RETURNING` statement.

### Observed Behaviour

SQLFluff reports an unparsable section

### How to reproduce

# Steps to reproduce
```bash
echo 'UPDATE my_table SET my_column = "SQLFluff rules!" RETURNING my_column;' | sqlfluff parse --dialect postgres -
```

# By Comparison - A similar statement that passes
```bash
echo 'INSERT INTO my_table (my_column) VALUES ("SQLFluff rocks!") RETURNING my_column;' | sqlfluff parse --dialect postgres -
```

### Dialect

Postgres

### Version

```
$ sqlfluff --version
> sqlfluff, version 0.13.0
```

```
$ python3 --version
> Python 3.9.12
```

Other Notes:
- I actually have sqlfluff installed [via Homebrew](https://formulae.brew.sh/formula/sqlfluff), but I have my doubts its related to this.
- For what it's worth, I see the same error when running this via the [included pre-commit hook](https://github.com/sqlfluff/sqlfluff/blob/main/.pre-commit-hooks.yaml) (which is how I found this)

### Configuration

None - just running plain `sqlfluff`

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
